R/make_theo_scores.R

```diff
@@ -20,12 +20,13 @@ make_theo_scores <- function(ValueTable, ParameterTable){
     # marking
     ValueTable <- ValueTable %>%
         dplyr::inner_join(ParameterTable , by = "Parameter_ID") %>%
+        dplyr::filter(!is.na(Value)) %>%
+        dplyr::filter(Value != "?") %>% 
```

Comment:
This shouldn't be necessary. `?` is specified as `null` value, see
https://github.com/grambank/grambank/blob/cb9adc433f7a4294d3f64569f91004b8a379ee58/cldf/StructureDataset-metadata.json#L83-L92
Thus, if properly read, `?` should already have been transformed to `NA`.

Reply:
The ValueTable that rcldf::cldf delivers treats the Value column from Grambank as a character vector, i.e. "?", "1" etc. ? are not transformed to NULL. They were never in grambank-analysed either, and I don't think it's a good idea to introduce as sometimes we are interested in the difference between NA and ?.

You can see this for yourself by going

`   str(rcldf::cldf(mdpath = "tests/testthat/fixtures/testdata/StructureDataset-metadata.json")$tables$ValueTable)`

Reply:
If you don't like this, please file an issue at https://github.com/SimonGreenhill/rcldf/issues

Reply:
As far as I can tell, the difference between the two variants of missing data are encoded in the CLDF data - if there's a `?` row for a (parameter, language) pair, it means "someone has tried to fill it in, but that it was not possible", when there's no row at all it means the other kind of missing data.

Reply:
> As far as I can tell, the difference between the two variants of missing data are encoded in the CLDF data - if there's a `?` row for a (parameter, language) pair, it means "someone has tried to fill it in, but that it was not possible", when there's no row at all it means the other kind of missing data.

genau. So, can you tell me again what you expect to happen with rcldf reading in the ValueTable? Did you expect ? values to me turned into NULL/NA and "pure" missing values to not be present rows in the long table at all? I think that distinction might be a bit hard to grasp, I would prefer if ? remains ?, hence https://github.com/grambank/pygrambank/issues/116

Reply:
I think "no row -> not investigated, hence not part of the data", "? -> NA" is the perfect way to encode this distinction. If a row has a non-null value for a categorical variable, that value must correspond to a category. But the category "NA for feature X" is ill-defined. `?` for language X may mean something completely different than `?` for language Y.

Reply:
I think "no row -> not investigated, hence not part of the data", "? -> NA" is the perfect way to encode this distinction. If a row has a non-null value for a categorical variable, that value must correspond to a category. But the category "NA for feature X" is ill-defined. `?` for language X may mean something completely different than `?` for language Y.

Reply:
Either way, as rcldf works _right now_ the highlighted line here for rgrambank::make_theo_scores() is correct.

Reply:
> I think "no row -> not investigated, hence not part of the data", "? -> NA" is the perfect way to encode this distinction. If a row has a non-null value for a categorical variable, that value must correspond to a category. But the category "NA for feature X" is ill-defined. `?` for language X may mean something completely different than `?` for language Y.

I disagree, especially given the already established habits in grambank-analysed scripts regarding making wide where non-existing rows are represented as NA cells in the wide format. I think it is better to be more explicit here and not make the ? -> NA/NULL transformation when reading in the long table ValueTable for Grambank. It makes the issue more apparent to a user and forces them to make an explicit decision, which I think is good.

Reply:
> Either way, as rcldf works _right now_ the highlighted line here for rgrambank::make_theo_scores() is correct.

Given that this is an rcldf and/or pygrambank issue @xrotwang , can we resolve this discussion here for this rgrambank function? At worst the code is unncessary given future versions of rcldf, but as it stands the code is good I think.